File: app/src/main/java/com/example/aichat/ChatApi.java
package com.example.aichat;

import com.google.gson.JsonObject;
import java.util.List;

import com.google.gson.annotations.SerializedName;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Url;

public interface ChatApi {
    @POST("chat/completions")
    Call<ChatResponse> chat(
            @Header("Authorization") String auth,
            @Header("Content-Type") String contentType,
            @Body ChatRequest request
    );

    @POST
    Call<ChatResponse> chatWithUrl(
            @Url String url,
            @Header("Authorization") String auth,
            @Header("Content-Type") String contentType,
            @Body ChatRequest request
    );

    class ChatRequest {
        public String model;
        public List<ChatMessage> messages;
        public boolean stream = false;
        public Integer n;
        @SerializedName("max_tokens")
        public Integer maxTokens;
        public Double temperature;
        @SerializedName("top_p")
        public Double topP;
        public List<String> stop;
        public Boolean thinking;
        public JsonObject reasoning;
        @SerializedName("providerOptions")
        public JsonObject providerOptions;
    }

    class ChatMessage {
        public String role;
        public String content;

        public ChatMessage(String role, String content) {
            this.role = role;
            this.content = content;
        }
    }

    class ChatResponse {
        public List<Choice> choices;
    }

    class Choice {
        public MessageDelta message;
    }

    class MessageDelta {
        public String content;
    }
}

File: app/src/main/java/com/example/aichat/ChatApi.kt
package com.example.aichat

import com.google.gson.JsonElement
import com.google.gson.JsonObject
import com.google.gson.annotations.SerializedName
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.Url

interface ChatApi {
    @POST("chat/completions")
    fun chat(
        @Header("Authorization") auth: String?,
        @Header("Content-Type") contentType: String,
        @Body request: ChatRequest
    ): Call<ChatResponse>

    @POST
    fun chatWithUrl(
        @Url url: String,
        @Header("Authorization") auth: String?,
        @Header("Content-Type") contentType: String,
        @Body request: ChatRequest
    ): Call<ChatResponse>

    class ChatRequest {
        @JvmField var model: String? = null
        @JvmField var messages: List<ChatMessage>? = null
        @JvmField var stream: Boolean = false
        @JvmField var n: Int? = null
        @SerializedName("max_tokens")
        @JvmField var maxTokens: Int? = null
        @JvmField var temperature: Double? = null
        @SerializedName("top_p")
        @JvmField var topP: Double? = null
        @JvmField var stop: List<String>? = null
        @JvmField var thinking: Boolean? = null
        @JvmField var reasoning: JsonObject? = null
        @SerializedName("response_format")
        @JvmField var responseFormat: JsonObject? = null
        @SerializedName("providerOptions")
        @JvmField var providerOptions: JsonObject? = null
    }

    class ChatMessage(
        @JvmField var role: String,
        @JvmField var content: String
    )

    class ChatResponse {
        @JvmField var choices: List<Choice>? = null
    }

    class Choice {
        @JvmField var message: MessageDelta? = null
    }

    class MessageDelta {
        @JvmField var content: JsonElement? = null
    }
}
